contracts/modules/UniswapManager.sol

```diff
@@ -0,0 +1,159 @@
+        }
+    }
+
+    function removeLiquidityFromUniswap(    
+        BaseWallet _wallet, 
+        address _poolToken, 
+        uint256 _amount
+    )
+        external 
+        onlyOwner(_wallet)
+        onlyWhenUnlocked(_wallet)        
+    {
+        address pool = uniswap.getExchange(_poolToken);
+        require(pool != address(0), "UM: The target token is not traded on Uniswap");
+        _wallet.invoke(pool, 0, abi.encodeWithSignature("removeLiquidity(uint256,uint256,uint256,uint256)",_amount, 1, 1, block.timestamp + 1));
+    }
+
+    /**
+     * @dev Computes the amount of tokens to swap and then pool given an amount of "major" and "minor" tokens,
+     * where there are more value of "major" tokens then "minor".
+     * @param _majorPoolSize The size of the pool in major tokens
+     * @param _minorPoolSize The size of the pool in minor tokens
+     * @param _majorAmount The amount of major token provided
+     * @param _minorAmount The amount of minor token provided
+     * @param _minorInMajor The amount of minor token converted to major (optional)
+     * @return the amount of major tokens to first swap and the amount of major and minor tokens that can be added to the pool after.
+     */
+    function computePooledValue(
+        uint256 _majorPoolSize,
+        uint256 _minorPoolSize, 
+        uint256 _majorAmount,
+        uint256 _minorAmount, 
+        uint256 _minorInMajor
+    ) 
+        internal 
+        view 
+        returns(uint256 _majorSwap, uint256 _majorPool, uint256 _minorPool) 
+    {
+        if(_minorInMajor == 0) {
+            _minorInMajor = getInputToOutputPrice(_minorAmount, _minorPoolSize, _majorPoolSize); 
+        }
+        _majorSwap = (_majorAmount.sub(_minorInMajor)).mul(1003).div(2000);
+        uint256 minorSwap = getInputToOutputPrice(_majorSwap, _majorPoolSize, _minorPoolSize);
+        _majorPool = _majorAmount.sub(_majorSwap);
+        _minorPool = _majorPool.mul(_minorPoolSize.sub(minorSwap)).div(_majorPoolSize.add(_majorSwap)) + 1;
+        uint256 minorPoolMax = _minorAmount.add(minorSwap);
+        if(_minorPool > minorPoolMax) {
+            _minorPool = minorPoolMax;
+            _majorPool = _minorPool.mul(_majorPoolSize.add(_majorSwap)).div(_minorPoolSize.sub(minorSwap)) + 1;
+        }
+        assert(_majorAmount >= _majorPool.add(_majorSwap));
+    }
+
+    /**
+     * @dev Computes the amount of output tokens that can be obtained by swapping the provided amoutn of input.
+     * @param _inputAmount The amount of input token.
+     * @param _inputPoolSize The size of the input pool.
+     * @param _outputPoolSize The size of the output pool.
+     */
+    function getInputToOutputPrice(uint256 _inputAmount, uint256 _inputPoolSize, uint256 _outputPoolSize) internal view returns(uint256) {
+        if(_inputAmount == 0) {
+            return 0;
+        }
+        uint256 inputWithFee = _inputAmount.mul(997);
```

Comment:
maybe clearer to call this `inputAfterFee` or `inputMinusFee`?

Reply:
Agreed and changed.